packages/block-editor/src/components/block-settings-menu/block-settings-dropdown.js

```diff
@@ -121,51 +121,40 @@ export function BlockSettingsDropdown( {
 			),
 		};
 	}, [] );
+	const isMatch = __unstableUseShortcutEventMatch();
 
 	const { selectBlock, toggleBlockHighlight } =
 		useDispatch( blockEditorStore );
 
 	const updateSelectionAfterDuplicate = useCallback(
-		__experimentalSelectBlock
-			? async ( clientIdsPromise ) => {
-					const ids = await clientIdsPromise;
-					if ( ids && ids[ 0 ] ) {
-						__experimentalSelectBlock( ids[ 0 ] );
-					}
-			  }
-			: noop,
+		async ( clientIdsPromise ) => {
+			if ( __experimentalSelectBlock ) {
+				const ids = await clientIdsPromise;
+				if ( ids && ids[ 0 ] ) {
+					__experimentalSelectBlock( ids[ 0 ] );
+				}
+			}
+		},
 		[ __experimentalSelectBlock ]
 	);
 
-	const updateSelectionAfterRemove = useCallback(
-		__experimentalSelectBlock
-			? () => {
-					const blockToSelect =
-						previousBlockClientId ||
-						nextBlockClientId ||
-						firstParentClientId;
+	const updateSelectionAfterRemove = useCallback( () => {
+		if ( __experimentalSelectBlock ) {
+			let blockToSelect = previousBlockClientId || firstParentClientId;
 
-					if (
-						blockToSelect &&
-						// From the block options dropdown, it's possible to remove a block that is not selected,
-						// in this case, it's not necessary to update the selection since the selected block wasn't removed.
-						selectedBlockClientIds.includes( firstBlockClientId ) &&
-						// Don't update selection when next/prev block also is in the selection ( and gets removed ),
-						// In case someone selects all blocks and removes them at once.
-						! selectedBlockClientIds.includes( blockToSelect )
-					) {
-						__experimentalSelectBlock( blockToSelect );
-					}
-			  }
-			: noop,
-		[
-			__experimentalSelectBlock,
-			previousBlockClientId,
-			nextBlockClientId,
-			firstParentClientId,
-			selectedBlockClientIds,
-		]
-	);
+			// Select the first block if there's no previous block nor parent block.
+			if ( ! blockToSelect ) {
+				blockToSelect = getBlockOrder()[ 0 ];
+			}
+
+			__experimentalSelectBlock( blockToSelect );
```

Comment:
The block selection is now changing rather than only focus shifting to the previous block in the list view. Is that intentional? I'm not sure how much more complicated it might be to implement, but I was wondering if ideally we wouldn't change the block selection at the same time 🤔

Reply:
My 2 cents are that block deletion should select the focused block **only if** there aren't any active selected blocks. This applies after the deletion happens so if a selected block gets deleted then we should select and focus the previous block too. I wonder if this makes sense to you 🤔 ? Or do you think we shouldn't update the selection no matter what? I think both make sense but probably there's something I'm missing!

Reply:
> so if a selected block gets deleted then we should select and focus the previous block too. I wonder if this makes sense to you

That makes sense to me! If a user deletes their block selection, it probably would be a bit odd to go from a block selection to no blocks selected at all, so in this case selecting the previous block sounds good to me 👍

Reply:
It now only update the selection if and only if the original selection is removed. 👍